config/dictionary_strings.rb

```diff
@@ -2031,6 +2031,12 @@
 _("Middleware Deployments")
 # TRANSLATORS: en.yml key: dictionary.table.middleware_deployments
 _("Middleware Deployments")
+# TRANSLATORS: en.yml key: dictionary.table.middleware_datasource
+_("Middleware Datasource")
+# TRANSLATORS: en.yml key: dictionary.table.middleware_datasource (plural form)
+_("Middleware Datasources")
+# TRANSLATORS: en.yml key: dictionary.table.middleware_datasources
+_("Middleware Datasources")
```

Comment:
You're not supposed to edit `config/dictionary_strings.rb` at all. That file is generated automatically and your
changes would be overwritten at next update.

All you really need to do is to edit `config/locales/en.yml` and add an entry under `dictionary.table.middleware_datasource` with value `Middleware Datasource` and that's it (i.e. don't add any plural forms, that part is also constructed
automatically.